src/staticswitcher/staticswitcher.c

```diff
@@ -1625,16 +1709,16 @@ switchPaintThumb (CompWindow		  *w,
 	    {
 		float xScale, yScale;
 
-		xScale = (float) ICON_SIZE / icon->width;
-		yScale = (float) ICON_SIZE / icon->height;
+		xScale = (float) staticswitcherGetPopupIconSize (s) / icon->width;
+		yScale = (float) staticswitcherGetPopupIconSize (s) / icon->height;
 
 		if (xScale < yScale)
 		    yScale = xScale;
 		else
 		    xScale = yScale;
 
-		sAttrib.xScale = (float) ss->previewWidth * xScale / PREVIEWSIZE;
-		sAttrib.yScale = (float) ss->previewWidth * yScale / PREVIEWSIZE;
+		sAttrib.xScale = (float) ss->previewWidth * xScale / staticswitcherGetPopupPreviewSize (s);
+		sAttrib.yScale = (float) ss->previewWidth * yScale / staticswitcherGetPopupPreviewSize (s);
 
```

Comment:
It would be a good idea to store the values of staticswitcherGetPopup*Size() calls in separate variables, then use those variables in the computational expressions.